### ISSUE TITLE
Support for HTTP reporter

### ISSUE THREAD

**Issue by iilyak** [sile/jaeger_passage#2]
Currently `jaeger_passage` supports only thrift over udp. This architecture requires jaeger-agent. Would you be interested in support for HTTP reporter which would connect directly to jaeger collector over port 14268? The implementation would require:
- a second reporter `jaeger_passage_http_reporer.erl`
- passing reporter module name to `jaeger_passage_reporter_sup.erl`
- making `thrift_protocol_binary:encode_struct/1` public

The tricky part is which http client library to use. There are tons of libraries and there is no good or bad choice here. I propose to pass a function callback in reporter options:

```
> Sampler = passage_sampler_all:new().
> ok = jaeger_passage:start_tracer(example_tracer, Sampler).

% Define http client callback
Client = fun(Url, Method, Headers, Body, ReporterOptions) ->
    User = proplists:get_value(user, ReporterOptions),
    Password = proplists:get_value(password, ReporterOptions),
    ibrowse:send_req(Url, Headers, Method, Body, [{basic_auth, {User,  Password}}])
end.

% Starts a root span.
> RootSpan = passage:start_span(example_root, [
    {tracer, example_tracer},
    {endpoint, "http://127.0.0.1:14268"},
    {http_client, Client}
]).
```

Are you interested in this contribution?

**Comment by sile** [comment]
Thank you for your proposal.
This feature seems useful, and I think your design choice about HTTP client is reasonable.
Of course, the contribution is very welcome!

**Comment by sile** [comment]
This issue was resolved by #3. Thanks @iilyak!
